test/Rule.js

```diff
@@ -417,6 +425,29 @@ test('merge with include and exclude not of array type', () => {
   });
 });
 
+test('merge with resolve', () => {
+  const rule = new Rule();
+
+  rule.merge({
+    resolve: {
+      alias: { foo: 'bar' },
+    },
+  });
+
+  rule.merge({
+    resolve: {
+      extensions: ['.js', '.mjs'],
+    }
+  })
+
+  expect(rule.toConfig()).toStrictEqual({
+    resolve: {
+      alias: { foo: 'bar' },
+      extensions: ['.js', '.mjs'],
+    },
+  });
+});
+
```

Comment:
I'm not sure that this assertion covers every case of `merge({ resolve })`. Please give me a nudge if there's still something missing @edmorley